File: 2144-maximum-difference-between-increasing-elements/maximum-difference-between-increasing-elements.c
int maximumDifference(int* nums, int numsSize) {
    int min = nums[0];
    int maxdiff = -1;
    for (int i = 0; i<numsSize;i++){
        if (nums[i]>min){
            int diff = nums[i]- min;
            if (diff > maxdiff)
            maxdiff = diff;
        }
        else
        min = nums[i];
    }
    return maxdiff;
}

File: 2144-maximum-difference-between-increasing-elements/maximum-difference-between-increasing-elements.java
class Solution {
    public int maximumDifference(int[] nums) {
        int min = nums[0];
        int maxdiff = -1;
        for (int i = 0;i<nums.length;i++){
            if (nums[i]> min){
                maxdiff = Math.max(maxdiff, nums[i]-min);
            }
            else
            min = nums[i];
        }
        return maxdiff;
    }
}